narwhals/_dask/dataframe.py

```diff
@@ -254,13 +254,20 @@ def join(
             )
 
         if how == "semi":
-            other_native = (
-                other._native_frame.loc[:, right_on]
-                .rename(  # rename to avoid creating extra columns in join
-                    columns=dict(zip(right_on, left_on))  # type: ignore[arg-type]
+            if isinstance(left_on, str) and isinstance(right_on, str):
+                other_native = (
+                    other._native_frame.loc[:, right_on]
+                    .to_frame(name=left_on)
+                    .drop_duplicates()  # avoids potential rows duplication from inner join
+                )
+            else:
```

Comment:
This fails for a case such as:

```py
...
right_on = ["abc"]
left_on = "abc"
```

I would suggest to convert the keys from `str` to `list[str]` at the beginning on the method and reason with them being lists.

Similarly below the check `if isinstance(left_on, str) and isinstance(right_on, str)` would fail for this such case and need to check them separately as we do in [pandas land](https://github.com/narwhals-dev/narwhals/blob/main/narwhals/_pandas_like/dataframe.py#L419)

Reply:
Thanks  for your review @FBruzzesi , I added a conversion from `str` to `list[str]` at the beginning of `join` for Dask